snuba/datasets/entity_subscriptions/validators.py

```diff
@@ -0,0 +1,55 @@
+from abc import abstractmethod
+from typing import Any, Mapping, Optional, Sequence, Type, Union, cast
+
+from snuba.query.composite import CompositeQuery
+from snuba.query.data_source.simple import Entity
+from snuba.query.logical import Query
+from snuba.query.validation.validators import (
+    NoTimeBasedConditionValidator,
+    SubscriptionAllowedClausesValidator,
+)
+from snuba.utils.registered_class import RegisteredClass
+
+
+class InvalidSubscriptionError(Exception):
+    pass
+
+
+class EntitySubscriptionValidator(metaclass=RegisteredClass):
+    @classmethod
+    def config_key(cls) -> str:
+        return cls.__name__
+
+    @classmethod
+    def get_from_name(cls, name: str) -> Type["EntitySubscriptionValidator"]:
+        return cast(Type["EntitySubscriptionValidator"], cls.class_from_name(name))
+
+    @abstractmethod
+    def to_dict(self, metadata: Mapping[str, Any]) -> Mapping[str, Any]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def validate(self, query: Union[CompositeQuery[Entity], Query]) -> None:
+        raise NotImplementedError
+
+
+class AggregationValidator(EntitySubscriptionValidator):
+    def __init__(
+        self,
+        max_allowed_aggregations: int,
+        disallowed_aggregations: Sequence[str],
+        required_time_column: Optional[str] = None,
+    ):
+        self.max_allowed_aggregations = max_allowed_aggregations
+        self.disallowed_aggregations = disallowed_aggregations
+        self.required_time_column = required_time_column
```

Comment:
> I assume this is avoiding some kind of circular import?

@evanh As discussed, added `required_time_column` as a param to AggregationValidator. This is used to fix the cyclic dependency.
Before: AggregationValidator uses entity factory to get required_time_column in entity. Causes cycle.
Now: `required_time_column` is specified in entity subscription config and used directly